src/backend/expungeservice/expunger/analyzers/time_analyzer.py

```diff
@@ -58,13 +60,37 @@ def _mark_all_acquittals_ineligible_using_mrd_date(expunger):
         for charge in expunger.acquittals:
             charge.set_time_ineligible('Recommend sequential expungement', eligibility_date)
 
+    @staticmethod
+    def _calculate_has_subsequent_charge(class_b_felony, charges):
+        other_charges = [charge for charge in charges if charge != class_b_felony]
+        for other_charge in other_charges:
+            if other_charge.acquitted():
+                date_of_other_charge = other_charge.date
+            else:
+                date_of_other_charge = other_charge.disposition.date
```

Comment:
I believe this logic is used in a lot of places.
```python
            if other_charge.acquitted():
                date_of_other_charge = other_charge.date
            else:
                date_of_other_charge = other_charge.disposition.date
```

It's probably a good time to extract this logic into a method and move it to the charge class. Or we can create a refactoring ticket to do that so that we can clean up the other places this similar logic is used. I'm pretty sure its used elsewhere although I haven't verified. -> even if its not used in other places, there is a method here that belongs on the charge class.

Reply:
Sure, refactoring this and also using a shared method for when we apply MRC and MRD rules, and checking a charge's own time eligibility, makes sense (and putting it on a different ticket sounds good), calling it something like ```Charge.date_that_applies_for_time_eligibility(self)```

Reply:
yeah, you may be right.

Reply:
I feel there is some logic in this private method that can be extracted, but maybe this is not the right time to do it.